openghg/store/base/_base.py

```diff
@@ -191,6 +191,27 @@ def add_datasources(self, uuids: Dict, data: Dict, metastore: TinyDB) -> None:
                 metastore.insert(meta_copy)
                 self._datasource_uuids[uid] = key
 
+    def add_single_datasource(self, uuid: Dict, data: Dict, metastore: TinyDB) -> None:
```

Comment:
This is handy but we could use `add_datasources` here. That's another function that's due a rework to make this kind of this easier.

Reply:
I couldn't get `add_datasources` to work, I think it would only when using the correct datasource_lookup -> assign_data -> add_datasources, as currently met data does not have the right format. Will fix to use the new store method rather than the existing METStore._store and then it should work